R/split_gene.R

```diff
@@ -1,57 +1,84 @@
 #' @title Split Gene Expression File Using Metadata
 #' @description A simple function to split gene expression data, using metadata treatment column
-#' @param geneExp (Object) Name of expression data frame object generated using read_gene() or filter_gene()
-#' @param metaPath (Character) Metadata file path
-#' @param simpleID (Character) Metadata column with samples/IDs found in gene expression data
-#' @param treatment (Character) Metadata column that designates control and treatment samples/IDs
+#' @param exp (Object) Name of expression data frame object generated using read_gene() or filter_gene()
+#' @param metadata_file (Character) Metadata file path
+#' @param id_column (Character) Metadata column with samples/IDs found in gene expression data
+#' @param treatment_column (Character) Metadata column that designates control and treatment samples/IDs
+#' @param expr_file_out (Character) Filename of the base group to write out results to.  No file will
+#' be written when NULL.
+#' @param comp_file_out (Character) Filename of the comparison group to write out results to.  No file
+#' will be written when NULL.
 #' @return Returns design and writes control (base) and treatment (comp) to working directory
 #' @author Walid Nashashibi (\url{https://github.com/wnashash/})
 #' @examples
 #' \dontrun{
 #' library(processNetZoo)
-#' exp_path <- system.file("extdata", "expression_test.csv", package = "processNetZoo", mustWork = TRUE)
+#'
+#' exp_path <- system.file("extdata",
+#'      "expression_test.csv",
+#'      package = "processNetZoo",
+#'      mustWork = TRUE
+#'      )
+#'
 #' expression <- read_gene(exp_path,'gene')
+#'
 #' generate_histogram(expression,'both')
+#'
 #' filtered <- filter_gene(expression,'sum')
-#' meta_path <- system.file("extdata", "metadata.csv", package = "processNetZoo", mustWork = TRUE)
+#'
+#' meta_path <- system.file("extdata",
+#'      "metadata.csv",
+#'      package = "processNetZoo",
+#'      mustWork = TRUE
+#'      )
+#'
 #' design <- split_gene(filtered,meta_path,'SUBJECT_ID','RECURRENCE_ANY')
 #' }
 #' @import data.table
 #' @import utils
 #' @export
 #'
-split_gene <- function(geneExp,metaPath,simpleID,treatment) {
+split_gene <- function(exp,
+                       metadata_file,
+                       id_column,
```

Comment:
Added "_column" suffix to make it clearer that users are looking to pass a data.frame name